### PR TITLE
fix(components): [cascader] remove equality check

### PR DESCRIPTION
**Pull request by Mookiepiece**
fix #20936

The use of `isEqual()` was introduced when this component was initially built on `elemeFE/element`. It makes the component not reflecting any array mutation methods.

In #934, we skipped the check when options updated by adding `force`.
In #10119, we added `cloneDeep` because deep cloning will pass the check when mutation happens. However, the deep clone was missed here:

https://github.com/element-plus/element-plus/blob/cfc661c62678901fdc4187262dfaa035369510c0/packages/components/cascader-panel/src/index.vue#L368-L371

When `checkedValue` changes due to a user's selection, `checkedValue` and `modelValue` are not deep cloned; they remain the same, which causes the issue.

By removing all of them, it works just fine for me.

---
##### ADD

在 elemeEF/element + Vue2 就已经存在，示例改编自官方站点的 Codepen。
[Vue2 Link](https://play.vuejs.org/#eNqNVVtv2zYU/isH2gDJgCMvXffi2Ua6NNgFy1YswV6qoqWpE4s1RQok5cQw/N97SMqyathF9SLqXL5z/ahd8qZp8k2LyTSZWW5E48Cia5tFoUTdaONgB1rd61Y5LGEPT0bXkJJD+muhekXG7FZxyEYwX8CuUADsmQkHESJLK+caO51MWtWsVznX9YQQbl5NSmGdP+afbToiQACulaUUYiZzKDVva1Qu5waZwzuJ/itLo0HnEz9yazh5nImF0euqFTev8utf8uvXEymWE6FKfPGRPUYfaKnLbS6UQvPH4/3fHtBXemrAmgZVmcXIMYtYscJneEc9Ehazj/NFl5pWUrOSwD6OCnUsUwq17kr6Rq3eqqt04JAblD4767YSbYXoYp5Dk8rgk7f5jma4isRXvGJy3TWGW3tSeYWs7Cq/rYQss0GsmN+GGbhnQlHQsAUANbpKl3Z6+AZom5Kqo1XpJQCuEjavdYkyt40UHLOfxtcR0j/7eNiP45v82dDf0MIaNcQLUNP3HzoH/+jGCWp6L9l36AE7flzegU+zUmxAlPMiofqLhNgRJFwya0m4lJqvgxhgZhumek2JtR+1YT46WTxQ/7kDprYgcUMj1E+H1CC7b6UTjUSioLci4Wg28XARGOUVZ5azEk3MfXMVCqUw4V0kUTztAEnenXpNY3Tj5Tuou1hTcKbFMfAK+frBGcGd3EYh9abz4xKZYUuJixktUJ9FSGu3C8H3oZGzZeucVnDDaYprChSnTYXHw2wSDXwDif2bxSfPB782t04b6vT/dBngi/Pk8otEdPGM8spslP9Y++smS3+gIaSk2gc2RS4dmjgHeD/YBL+TsqUq01UrSkwHG0FcYUtaE0h/P1Xtx8NtOmIQgRqtaEXO49yeU1/CMmh1a/iFlP47q+WedgbV9KTGIyp7ac2J0xH0jVdCn2W8+S7lecS0a3S8ugT6ELTwiHUjacLfC0pks5cg36IVKwVvOz4G7nwN+2HQXzoTS8JFS2uVjBNnaSWexIrudq3oxxYyKBI/PCHR/HvgRH8pEaul1M9/BVmgw0EeWHFG/tm+eFmRvKMxotnQhvc6x8wKXVTfPfxD2zxQEldaSdbfUPrRyzbeF97st1aVlPbALmT7Z/i3CrV6tHeeMPZQlE80XGrBvkjo/+pHfqn0Y7o/56+DHxEr2X8ByPmRvQ==)

---
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/cascader-panel/src/index.vue [hunk 1]
```diff
@@ -233,8 +228,8 @@ const syncCheckedValue = (loaded = false, forced = false) => {
         store?.getNodeByValue(val as CascaderNodeValue, leafOnly)
       )
     ) as Node[]
-    syncMenuState(nodes, forced)
-    checkedValue.value = cloneDeep(modelValue)
+    syncMenuState(nodes, false)
+    checkedValue.value = modelValue
```
Comment (Dsaquel) [inline packages/components/cascader-panel/src/index.vue]: >In https://github.com/element-plus/element-plus/pull/10119, we added cloneDeep because deep cloning will pass the check when mutation happens. However, the deep clone was missed

From what i seen in https://github.com/element-plus/element-plus/pull/10119
It's surely to avoid shallow copy the prop `modelValue`, it's used to to prevent the `isEqual` to pass.

I didn't test all cases but there are surely edge case that are may not been handled in tests.

Comment (Mookiepiece) [reply packages/components/cascader-panel/src/index.vue]: Since I've removed `isEqual()`, any concern related to `isEqual` won't be an issue.

Comment (Mookiepiece) [reply packages/components/cascader-panel/src/index.vue]: 我相信那个 PR 和我要解决的是同一个问题，因为点进去那里二楼的 playground 你可以看到代码里 ` // 没有用` 和 `// 有用` 这两个注释。

我把 `isEqual` 删掉了也许会有漏掉的用例！但把 `cloneDeep` 去掉在我看来没有副作用。

Comment (Mookiepiece) [reply packages/components/cascader-panel/src/index.vue]: FYI, I found continuing removing another `isEqual` for `modelValue` and `checkedValue` works fine, as `deepClone` for `modelValue` has been removed.

OLD: 

https://github.com/element-plus/element-plus/blob/cfc661c62678901fdc4187262dfaa035369510c0/packages/components/cascader-panel/src/index.vue#L365-L373

NEW:

```ts
watch(checkedValue, (val) => {
  if (val === props.modelValue) return // Update
  emit(UPDATE_MODEL_EVENT, val)
  emit(CHANGE_EVENT, val)
})
```

File: packages/components/cascader-panel/src/index.vue [hunk 2]
```diff
@@ -233,8 +228,8 @@ const syncCheckedValue = (loaded = false, forced = false) => {
         store?.getNodeByValue(val as CascaderNodeValue, leafOnly)
       )
     ) as Node[]
-    syncMenuState(nodes, forced)
-    checkedValue.value = cloneDeep(modelValue)
+    syncMenuState(nodes, false)
```
Comment (Mookiepiece) [inline packages/components/cascader-panel/src/index.vue]: With the removal of `isEqual`, calling `syncMenuState()` with `reserveExpandingState = false` makes the menu always flashes to the first panel after any user intactions

Reproduction with current code: 
- Check `1st > 1st > 1st`
- Check `1st > 2nd > 1st` 
- Result: the menu redirects to `1st > 1st > 1st`.

I have to fix this

File: packages/components/cascader-panel/src/index.vue (hunk 1 above)
Comment (Dsaquel) [reply packages/components/cascader-panel/src/index.vue]: i think `props.modelValue` was shallow copied into `checkdValue`, `cloneDeep` do create another reference.
That said judging by references (`val === props.modelValue`) might not be sufficient.

In https://github.com/element-plus/element-plus/pull/10119 there is no test implemented, i'm afraid such changes will make regressions :thinking: .

**Comment by Mookiepiece** [comment]
I found an odd fact that `checkedValue` is unused.

```mermaid
flowchart LR
subgraph "Main"
        A["UI Checkbox"]
        A -- "(1)" --> B(("checkedValue"))
        B -- "(2) watcher" --> C(("modelValue"))
        C --  "(3) watcher" --> D["syncCheckedValue"]
        D --  "(4) async" --> B
        B --> D
end
        E["initStore (props.options)"] -- "(5)" --> D
        D -- "(6) async" --> F["syncMenuState"]
```

Let's walk through all possible triggers:
| Trigger | Data Flow |
|--|--|
| User Clicks | 1, 2, 3, 6 & 4 |
| Assignment to `modelValue` | 3, 6 & 4 |
| `options` prop changed | 5, 6 & 4 |

**Still don't know what use case will make the `modelValue` and `checkedValue` be different thing in this table.**

But we got an circle(2, 3, 4).

---

This is my revision, after the removal of `checkedValue`, the data flow is much clear.

```mermaid
flowchart LR
subgraph "Main"
        A["UI Checkbox"]
        A -- "(2)" --> C(("modelValue"))
        C --  "(3) watcher" --> D["syncMenu (prev. syncCheckedValue)"]

end
        E["initStore (props.options)"] -- "(5)" --> D
        D -- "(6) async" --> F["syncMenuState"]
```

| Trigger | Data Flow |
|--|--|
| User Clicks |  2, 3, 6 |
| Assignment to `modelValue` | 3, 6 |
| `options` prop changed | 5, 6 |

---

On the other hand, as I commentted in #20936 , we'll encounter a UX problem.

**Each time you update the `modelValue` with javascript, the menu list redirects to the first selected option.**

We even have existing test cases that covers this "optimistic redirects".

Since `3, 6` is a subset of `2, 3, 6`, I have to use a ugly `nextTick(() => manualChecked = false)` to distinguish between `236` and `36` to ensure assignments to `modelValue` during `onChange` will not make the menu list redirects.

It's on you to decide whether we should mark the issue as not planned.

**Comment by Mookiepiece** [comment]
lack of attention, i'll close it.
